### PR TITLE
feat(container): delete a deployment

### PR DESCRIPTION
**Pull request by joshuachp**

### CODE REVIEW

File: edgehog-device-runtime-containers/src/events.rs [hunk 1]
```diff
@@ -72,6 +72,7 @@ pub(crate) enum EventStatus {
     Starting,
     Stopping,
     Updating,
+    Deleteing,
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/events.rs:75]:
```suggestion
    Deleting,
```

File: edgehog-device-runtime-containers/src/events.rs [hunk 2]
```diff
@@ -81,6 +82,7 @@ impl Display for EventStatus {
             EventStatus::Starting => write!(f, "Starting"),
             EventStatus::Stopping => write!(f, "Stopping"),
             EventStatus::Updating => write!(f, "Updating"),
+            EventStatus::Deleteing => write!(f, "Deleting"),
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/events.rs:85]:
```suggestion
            EventStatus::Deleting => write!(f, "Deleting"),
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: edgehog-device-runtime-containers/src/properties/volume.rs [hunk 3]
```diff
@@ -23,25 +23,26 @@ use uuid::Uuid;
...
-        Self { id, created }
+    pub(crate) fn new(id: &'a Id) -> Self {
+        Self { id }
     }
 }
 
 #[async_trait]
 impl AvailableProp for AvailableVolumes<'_> {
+    type Data = bool;
+
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/properties/volume.rs:45]: Is it still valid?

File: edgehog-device-runtime-containers/src/service/node.rs [hunk 4]
```diff
@@ -281,6 +325,28 @@ impl Node {
...
+    where
+        D: Client + Sync + 'static,
+    {
+        let resource = self.resource.as_ref().ok_or(ServiceError::Missing {
+            id: self.id,
+            ctx: "up",
+        })?;
+
+        if resource.state == State::Received {
+            warn!("removind resource {} which is not stored", self.id);
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/service/node.rs:339]:
```suggestion
            warn!("removing resource {} which is not stored", self.id);
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Review by rgallor — APPROVED** [review]

File: edgehog-device-runtime-containers/src/properties/volume.rs (hunk 3 above)
Comment (joshuachp) [reply edgehog-device-runtime-containers/src/properties/volume.rs:45]: Yes, I changed the AvailableProp trait to support the unset. Now it requires an associated type for the property to send

**Review by harlem88 — APPROVED** [review]

**Pull request merged**